functions/src/create-notification.ts

```diff
@@ -185,149 +194,253 @@ export const createNotification = async (
     await notifyContractCreator(userToReasonTexts, sourceContract)
   }
 
-  await createUsersNotifications(userToReasonTexts)
+  await sendNotificationsIfSettingsPermit(userToReasonTexts)
+}
+
+export const getDestinationsForUser = async (
+  userId: string,
+  reason: notification_reason_types | keyof notification_subscription_types
+) => {
+  const privateUser = await getPrivateUser(userId)
```

Comment:
You can leave it this way, but generally it's better fetch the private user separately, and have this function with logic be non-async. 
There's a number of benefits including ability to reuse the function more (even on the client).